File: Java-edition-2nd/MyApplication/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.content.DialogInterface;
import android.content.Intent;
import android.media.Image;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private EditText editText;
    private ImageView imageView;
    private ProgressBar progressBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button button = (Button)findViewById(R.id.button);
        editText = (EditText)findViewById(R.id.text_edit);
        imageView = findViewById(R.id.image_view);
        final TextView text_view = (TextView)findViewById(R.id.text_view);
        progressBar = (ProgressBar)findViewById(R.id.progress_bar);

        imageView.setVisibility(View.GONE);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String inputText = editText.getText().toString();
                if (!inputText.equals(""))
                    Toast.makeText(MainActivity.this, inputText, Toast.LENGTH_SHORT).show();
                text_view.setText(inputText);
                progressBar.setProgress(progressBar.getProgress() + 10);
//                if (progressBar.getVisibility() == View.GONE)
//                    progressBar.setVisibility(View.VISIBLE);
//                else
//                    progressBar.setVisibility(View.GONE);
                AlertDialog.Builder dialog = new AlertDialog.Builder(MainActivity.this);
                dialog.setTitle("This is a dialog");
                dialog.setMessage("Something important");
                dialog.setCancelable(false);
                dialog.setPositiveButton("OK", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {

                    }
                });
                dialog.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {

                    }
                });
                dialog.show();
            }
        });

    }
}

File: Chapter_4/UIWidgetTest/app/src/main/java/com/example/uiwidgettest/MainActivity.kt
package com.example.uiwidgettest

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), View.OnClickListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button.setOnClickListener(this)
        button_changePic.setOnClickListener(this)
        button_changeProgress.setOnClickListener(this)
        button_addProgress.setOnClickListener(this)
        button_alertDialog.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.button -> {
                val intputText = editText.text.toString()
                if (intputText != "") {
                    Toast.makeText(this, intputText, Toast.LENGTH_SHORT).show()
                }
            }
            R.id.button_changePic -> {
                imageView.setImageResource(R.drawable.img_2)
            }
            R.id.button_changeProgress -> {
                if (progressBar.visibility == View.VISIBLE) {
                    progressBar.visibility = View.GONE
                } else {
                    progressBar.visibility = View.VISIBLE
                }
            }
            R.id.button_addProgress -> {
                progressBar2.progress += 10
            }
            R.id.button_alertDialog -> {
                AlertDialog.Builder(this).apply {
                    setTitle("This is Dialog")
                    setMessage("Something important")
                    setCancelable(false)
                    setPositiveButton("OK") { dialog, which ->
                    }
                    setNegativeButton("Cancel") { dialog, which ->
                    }
                    show()
                }
            }
        }
    }
}